keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -94,18 +96,46 @@ own APIs and controllers or adopt third party implementations. Each
 implementation splits the ecosystem, making it harder for higher level systems
 for Job queueing or workflows to support all of them.
 
+Additionally, the Pods within a Job can't easily address and communicate with
+each other, making it hard to run tightly coupled parallel Jobs using the Job
+API.
+
+Third-party operators cover these use cases by defining their own APIs, leading
+to fragmentation of the ecosystem. The operators use mainly two networking
+patterns: (1) fronting each index with a Service or (2) creating Pods with
+stable hostnames based on their index.
+
+The problem with using a Service per index is twofold:
```

Comment:
If using a headless service, those aren't really a problem right?
1. you can have the pod IP directly from DNS (this is what headless does in DNS)
2. IIRC we don't program kube-proxies for headless services

Yes with selector headless service create an endpoint, so there is some overhead, but doesn't seem to be that important argument.

Reply:
Right, it depends on what time of service the user chooses.
The second point still applies though: there is one service and one endpoint object for each index, instead of one for the job.

But one of the intentions of these paragraphs is that the fragmentation leads to inefficiencies due to poor choose of APIs.
Still, maybe I can remove the first point. WDYT?